src/address_operands.rs

```diff
@@ -74,6 +74,11 @@ pub fn address_operands_read(vm: &mut VMState, opcode: &Opcode) -> (TaggedValue,
                 }
             }
         }
+    };
+    if opcode.swap_flag {
+        (op2, op1)
+    } else {
+        (op1, op2)
```

Comment:
I'm not sure if this is how we should do it, I think not all opcodes support swapping, maybe this should be implemented in each opcode. (Please check this, maybe its fine)

Reply:
As you say, some functions, like `Mul`, don't support the `.s`, this is taken care of by the compiler, which forbids bulding this kind of code:
```asm
	mul.s 3, r1, r3, r4
```
throws:
```bash
 error: assembly parsing: can not parse text section element: cannot parse lines: {6: ("mul.s 3, r1, r3, r4", UnknownArgument("Mul instruction contains unknown modifiers: {\"s\"}"))}
 ```